### ISSUE TITLE
Cannot find module 'minimatch'

### ISSUE THREAD

**Issue by adamreisnz** [Unitech/pm2#3157]
## What's going wrong?
pm2 won't install correctly, seems to have a dependency missing;

```
Error: Cannot find module 'minimatch'
    at Function.Module._resolveFilename (module.js:527:15)
    at Function.Module._load (module.js:476:23)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/lib/node_modules/pm2/node_modules/glob/glob.js:45:17)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
```

## How could we reproduce this issue?
npm i -g pm2

## Supporting information

Please run the following command (available on PM2 >= 2.6)

```
$ pm2 report
```
Doesn't work because it throws the same error for this command.

## Additional information
Tried to reinstall several times, to no avail.

OS: Raspbian Lite Stretch (Debian)
Node: 8.5.0
Npm: 5.3.0
pm2: 2.7.0

Log of install (without errors):

```
sudo npm i -g pm2
/usr/local/bin/pm2 -> /usr/local/lib/node_modules/pm2/bin/pm2
/usr/local/bin/pm2-dev -> /usr/local/lib/node_modules/pm2/bin/pm2-dev
/usr/local/bin/pm2-docker -> /usr/local/lib/node_modules/pm2/bin/pm2-docker
/usr/local/bin/pm2-runtime -> /usr/local/lib/node_modules/pm2/bin/pm2-runtime
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.2 (node_modules/pm2/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm"})

+ pm2@2.7.0
added 251 packages in 36.54s
```

**Comment by adamreisnz** [comment]
After a reboot of the device and trying again it worked ok this time. Any thoughts of what could have caused this malfunctioning installation?

**Comment by nazarewk** [comment]
@adamreisnz i am pretty sure this is issue with npm 5.3.0, for me it didn't work for installation in a single `npm i -g npm pm2` on `node:8.5.0` docker, but `npm i -g npm@5.4.2 && npm i -g pm2` worked just fine.

**Comment by vmarchaud** [comment]
Indeed, npm might have simply not installed minimatch

**Comment by adamreisnz** [comment]
That's possible, but I didn't upgrade npm after reboot. I thought maybe the order in which I installed node and npm made a difference. I used to install (upgrade) npm, n, pm2 and yarn at the same time, then use n to get the latest node version. I changed that around to install n and npm first, then install the latest node, and only then install pm2 and that's working ok now for new installations.

Probably a fluke issue though 👍
